### PR TITLE
[FIX] purchase_landed_cost: error on deleting cost lines

### PR DESCRIPTION
**Pull request by cubells**
Before this pr:
If user try to delete some cost line and this cost line is referenced in any expense line into the field 'affected lines', a not _sensitive_ error is shown to user.

After this pr:
Error is catched and an _clear_ message is shown to user.

cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can't this be handled through ondelete="cascade"?

**Comment by cubells** [comment]
@pedrobaeza 
cost_lines has `ondelete` parameter: https://github.com/Tecnativa/purchase-workflow/blob/11.0-product_landed_cost_fix_error/purchase_landed_cost/models/purchase_cost_distribution.py#L120

and error is raised when deleting a cost line without remove reference in expense lines.

**Comment by pedrobaeza** [comment]
That ondelete doesn't serve. It should be on the `many2one` part, not on the `one2many`

**Comment by cubells** [comment]
I've tested `ondelete` on`affected_lines` field and it does not work, but `affected_lines` field is Many2many.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@Roodin were you finally using this?

File: purchase_landed_cost/__manifest__.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
-              "Joaquín Gutierrez,"
-              "Odoo Community Association (OCA)",
+    'version': '10.0.2.0.0',
+    "author": u"OdooMRP team,"
+              u"AvanzOSC,"
+              u"Tecnativa,"
+              u"Joaquín Gutierrez,"
+              u"Odoo Community Association (OCA)",
     'category': 'Purchase Management',
     'website': 'http://www.odoomrp.com',
```
Comment (rafaelbn) [inline purchase_landed_cost/__manifest__.py]: Change to https://github.com/OCA/purchase-workflow
```suggestion
    'website': 'https://github.com/OCA/purchase-workflow',
```
As said in https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11

**Review by rafaelbn — APPROVED** [review]

**Comment by rafaelbn** [comment]
Are we going to improve  codecov/patch — 76.47% of diff hit (target 81.1%) ?

Can we merge ?

**Comment by pedrobaeza** [comment]
Yes, merging, the coverage is good on the project and module, but as this is reducing code, that's why it's red.

**Pull request merged**